### ISSUE TITLE
the error in change  RPN model:conv5_3  to conv4_3

### ISSUE THREAD

**Issue by dlongry** [endernewton/tf-faster-rcnn#167]
when I directly delete pool4 and conv5 layers, the error show as follows:
![Uploading 2017-07-22 20:03:47屏幕截图.png…]()
then, in lib/nets/network.py ,I modified feat_stride = 16 to feat_stride = 8
the train is run,but loss=nan ,  error as follows:
 
![2017-07-22 20 08 48](https://user-images.githubusercontent.com/16435037/28490979-986de12e-6f19-11e7-97c3-d18464b1994c.png)
Do you have any solution for this error,thanks,I want to change RPN network for my own. @ShijunK

**Comment by endernewton** [comment]
you may have to change a couple of other things as well. the current code supports feat stride 16. it also computes RoI with 1/16 and so on.

**Comment by dlongry** [comment]
Thanks to your answer, but I can't find  where to modify ROI Steps. Could you tell me where to modify ROI to 1/8?  Thanks again! @endernewton

**Comment by dlongry** [comment]
I modified self._feat_stride and  lr to solve this problem. It seems that it can work. but at the beginning, show a log as follows:
![1](https://user-images.githubusercontent.com/16435037/28917014-b9748b46-7876-11e7-8a90-968815bfb528.png)
what's the meaning of it ?

**Comment by TheRevanchist** [comment]
Having the same problem here, and weirdly enough, at times it gives NaN, at times no. I also changed line 248 in config.py to

    __C.DEDUP_BOXES = 1. / 16.

**Comment by dlongry** [comment]
I have solved this problem. I change the learing rate of  lib/model/config.py  and  modify the lib/net/network feat_stride and if you change you input image size,you should also change the base_size of the lib/layer_uitls/generate_anchor.py. After these, it  worked. I hope these tips can help you.

**Comment by TheRevanchist** [comment]
So, as far as I can understand you changed:

self._feat_stride and self._feat_compress from 16 to 8 (in network.py)
base_size from 16 to 8 in generate_anchors.py
DEDUP_BOXES from 16 to 8 in config.py
lower learning rate in config.py

I have done the same too, but still getting NaNs. I am using some other dataset and pretrained net, so the problem might be somewhere else.

Thanks!

**Comment by yaoqi-zd** [comment]
@TheRevanchist Hi, did you find any usages of DEDUP_BOXES in the source code? I have tried but failed to find any. if this parameter is unused, then we may ignore it.
